61 story1 13 view number of free rooms in a building

mmc03-ucb opened the pull request:
## Add RoomStatusService for Free Room Count

**Closes:** #61

Implements `RoomStatusService` to count free rooms via `/api/rooms/status`.

- Added `LiveRoomStatusService` with `getFreeRoomCount(buildingId:)`
- Added unit tests with `MockHTTPClient`
- Updated package dependencies

```swift
let count = await service.getFreeRoomCount(buildingId: "K-J17")
```

xleonx0x commented on ios/Rooms/Sources/RoomServices/RoomStatusService.swift:
```diff
@@ -0,0 +1,74 @@
+
+  // MARK: Public
+
+  public func getFreeRoomCount(buildingId: String) async -> Result<Int, RoomStatusError> {
```
This would be O(nm) since we have to call this for each building and then loop through each room to filter. I asked Michael to work on an endpoint so this logic is done in the backend.

xleonx0x reviewed (commented):
We're considering a redesign on the backend endpoint which would make this redundant. i.e the get buildings endpoint will come with this as a response.

xleonx0x commented:
@mmc03-ucb Michael modified the `/buildings` route to include `numberOfAvaliableRooms` as a field. Currently our offline implementation just fetches it offline, there should be networking added in the interactor that merges both offline and `numberOfAvaliableRooms` via a network request. Otherwise if theres a connectivity error it should simply return the offline data.

mmc03-ucb commented:
Removed deprecated RoomStatusService, moved RoomStatusLoader to Rooms package to avoid circular dependencies.
Renamed LocationService to LiveLocationService, LocationServiceProtocol to LocationService, RoomStatusResponse to RemoteRoomStatus.

xleonx0x reviewed (approved):
lgtm
